mindsdb/integrations/handlers/s3_handler/s3_handler.py

```diff
@@ -157,7 +157,10 @@ def _connect_duckdb(self, bucket):
         # detect region for bucket
         if bucket not in self._regions:
             client = self.connect()
-            self._regions[bucket] = client.get_bucket_location(Bucket=bucket)["LocationConstraint"]
+            location = client.get_bucket_location(Bucket=bucket)["LocationConstraint"]
```

Comment:
maybe you can do ` client.get_bucket_location(Bucket=bucket).get("LocationConstraint", None)` and maybe wrap this in try and catch block